Todo_TS/src/TodoList.ts

```diff
@@ -0,0 +1,56 @@
+import { ITodoList, ITodo, ICreateTodo } from '../types/todoTypes';
+import { createTodoElement } from '../utils/createElement';
+
+export default function TodoList({
+  $target,
+  initialState,
+  handleComplete,
+  handleDelete,
+}: ITodoList) {
+  const $listWrap = document.createElement('ul');
+  $target && $target.appendChild($listWrap);
+
+  let state: ITodo[] = initialState;
+  let isInit = false;
+
+  const setState = (nextState: ITodo[]) => {
+    // console.log('TodoList에서 nextState', nextState);
+    state = nextState;
+
+    if (!isInit) {
+      $listWrap.innerHTML = '';
+    }
+
+    state &&
+      state.map(({ text, isCompleted, idx }: ITodo) => {
+        const todoDiv = document.createElement('div');
+        todoDiv.className = 'list_div';
+
+        // prettier-ignore
+        const todoLi = createTodoElement({$target: todoDiv,element: "li",idx,text});
+        if (isCompleted) {
+          todoLi.className = 'done';
+        }
+        todoLi.addEventListener('click', (e: any) => {
+          const idx = e.target.dataset.idx;
```

Comment:
```suggestion
          const idx: string= e.target.dataset.idx;
```
P5: `idx`에 `string` 타입이 필요할 거 같습니다!